internal/driver/fetch.go

```diff
@@ -357,7 +360,7 @@ func fetch(source string, duration, timeout time.Duration, ui plugin.UI) (p *pro
 		f, err = fetchURL(sourceURL, timeout)
 		src = sourceURL
 	} else {
-		f, err = os.Open(source)
+		f, err = profileProtoReader(source)
```

Comment:
Have profileProtoReader returns a *profile.Profile

Otherwise it isn't really a "proto reader"